support/util/util.go

```diff
@@ -458,3 +458,109 @@ func PredicatesForHostedClusterChildResourcesAnnotationScoping(r client.Reader)
 	}
 	return predicate.NewPredicateFuncs(filter)
 }
+
+// PredicatesForNodepoolAnnotationScoping returns predicate filters for all event types that will ignore incoming
+// event requests for nodepool resources in which their owning hostedcluster resource doesn't have a scope annotation
+// that matches what is specified in the HOSTEDCLUSTERS_SCOPE_ANNOTATION env var.  If not defined or empty, the default behavior
+// is to accept all nodepool events in which the owning hostedcluster resource does not have a corresponding scope annotation defined.
+// The ENABLE_HOSTEDCLUSTERS_ANNOTATION_SCOPING env var must also be set to "true" to enable the scoping feature.
+func PredicatesForNodepoolAnnotationScoping(r client.Reader) predicate.Predicate {
+	hcAnnotationScopingEnabledEnvVal := os.Getenv(EnableHostedClustersAnnotationScopingEnv)
```

Comment:
nit: let's prefer to read these values at startup or as close to startup in `main()` as we can and pass them via normal function parameters to aid in understanding control flow